Assets/Plugins/Source/Core/Utility/JsonUtility.cs

```diff
@@ -22,21 +22,106 @@
 
 namespace PlayEveryWare.EpicOnlineServices.Utility
 {
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using System;
+    using UnityEngine;
+
     public static class JsonUtility
     {
+        /// <summary>
+        /// Attempts to parse the given JSON string. If problems are encountered, an error will be logged and the exception will continue.
+        /// </summary>
+        /// <param name="json">The JSON string to validate.</param>
+        private static void ValidateJson(string json)
+        {
+            try
+            {
+                JToken.Parse(json);
+            }
+            catch (JsonReaderException ex)
+            {
+                Debug.LogError($"Invalid JSON: {ex.Message}");
+#if UNITY_EDITOR
+                throw;
+#endif
+            }
+            catch (Exception ex)
+            {
+                Debug.LogError($"An error occurred: {ex.Message}");
+#if UNITY_EDITOR
+                throw;
+#endif
+            }
+        }
+
+        /// <summary>
+        /// Determines if the given json string is valid json or not. If JSON is invalid, errors will be logged, but exceptions will be suppressed.
+        /// </summary>
+        /// <param name="json">The JSON string to validate.</param>
+        /// <returns>True if json is valid, false otherwise.</returns>
+        private static bool IsJsonValid(string json)
+        {
+            try
+            {
+                ValidateJson(json);
+                return true;
```

Comment:
Note that as a consequence of the decision to not throw exceptions during non-editor runtime, the `IsJsonValid` function will always return `true` when not in Editor. This may or may not be desirable behavior.